API_Movie_Call.py

```diff
@@ -27,16 +27,23 @@
 # URL for the images, w500 is not part of the base url, but it is the max width for a poster.
 movie_images_url = 'https://image.tmdb.org/t/p/w500/'
 
-def main():
+def movie_call(date):
+    year, month, day = split_date(date)
     query =set_api_discover_params(year, month, day, key)
     data = api_movie_request(movie_discover_url, query)
     movie_raw = get_movie_results(data)
-    movie_title, movie_overview, movie_poster_path = get_movie_details(movie_raw)
-    movie_img_link = get_movie_image(movie_poster_path)
+    # movie_title, movie_overview, movie_poster_path = get_movie_details(movie_raw)
+    # movie_img_link = get_movie_image(movie_poster_path)
+    movie_info = get_movie_details(movie_raw)
 
-    print(movie_title)
-    print(movie_overview)
-    print(movie_img_link)
+    # print(movie_title)
+    # print(movie_overview)
+    # print(movie_img_link)
+    return movie_info
+
+def split_date(date):
+    date_split = date.split('-')
```

Comment:
Python has list destructuring so you could also write 

year, month, day = date.split('-')

Reply:
The dates are in the YYYY-MM-DD format? Consider using the date library

```
from datetime import date

d = date.fromisoformat('2011-02-01')   # no problem
# access year, month, day of the date object
print(d.year, d.month, d.day) # 2011 02 01  
# raises an error if the date is invalid - so you'll know if the user has somehow managed to send an invalid date
d2 = date.fromisoformat('2011-99-99')  # ValueError
```

And can check if the date is in the past or not,

```
today = date.today()
if d > today:
    print('date is in the future')
```

Don't trust your user to only send requests with valid dates.